app/products/calls/utils.ts

```diff
@@ -105,6 +105,30 @@ export function isCallsCustomMessage(post: PostModel | Post): boolean {
     return Boolean(post.type && post.type === Post.POST_TYPES.CUSTOM_CALLS);
 }
 
+export function isBoRPost(post: PostModel | Post): boolean {
+    return Boolean(post.type && post.type === Post.POST_TYPES.BURN_ON_READ);
+}
+
+export function isUnrevealedBoRPost(post: PostModel): boolean {
+    return isBoRPost(post) && Boolean(!post.metadata?.expire_at);
+}
+
+export function isOwnBoRPost(post: PostModel, currentUser?: UserModel): boolean {
+    return isBoRPost(post) && Boolean(currentUser && post.userId === currentUser.id);
+}
+
+function isBoRPostExpiredForMe(post: PostModel): boolean {
+    return Boolean(post.metadata?.expire_at) && post.metadata!.expire_at! < Date.now();
+}
+
+function isBorPostExpiredForAll(post: PostModel): boolean {
+    return Boolean(post.props?.expire_at) && ensureNumber(post.props!.expire_at!) < Date.now();
+}
+
+export function isExpiredBoRPost(post: PostModel): boolean {
+    return isBoRPost(post) && (isBorPostExpiredForAll(post) || isBoRPostExpiredForMe(post));
+}
+
```

Comment:
Why are this functions is the calls utils? it kind of makes no sense at all, perhaps move under `app/utils/bor/index.ts` ?